src/codegen/case_translator.cpp

```diff
@@ -58,8 +58,6 @@ codegen::Value CaseTranslator::DeriveValue(CodeGen &codegen,
     }
     when.EndIf(merge_bb);
   }
-  // Jump to the merging block from the internal If merging block
-  codegen->CreateBr(merge_bb);
```

Comment:
The original implementation will cause an LLVM compilation error if we change `DerivedValue()` for constant VARCHAR (I just added a dummy function call, but I guess a load instruction might trigger the same error). I made a branch containing only that change as a demo: https://github.com/phisiart/peloton-p3/tree/case. The failing test case is "case_sql_test".

I'm not sure whether the modification here is correct - though it passed tests. Could you take a look at the error above and figure out what's the correct way of handling this?